fs-base/src/main/java/com/palantir/giraffe/file/MoreFiles.java

```diff
@@ -493,71 +492,102 @@ public static boolean isEmpty(Path path) throws IOException {
     }
 
     /**
-     * Given a {@code Path} representing a directory, retrieve the list of files
-     * and directories contained in that {@code Path} as a {@code List}. The
-     * {@code Path} objects are obtained as if by resolving the name of the
-     * directory objects against the given {@code Path}.
+     * Gets a list of the files and directories in the given directory. The
+     * returned paths are {@linkplain Path#resolve(Path) resolved} against the
+     * given path.
      *
-     * @param directoryPath The {@code Path} to get the directory entries for
+     * @param dir the {@code Path} to list
      *
-     * @return A list of {@code Path} objects representing the directory entries
+     * @throws IOException if an I/O error occurs while listing entries
+     */
+    public static List<Path> listDirectory(Path dir) throws IOException {
+        try (DirectoryStream<Path> entries = Files.newDirectoryStream(dir)) {
+            return Lists.newArrayList(entries);
+        }
+    }
+
+    /**
+     * Gets a list of the files in the given directory. The returned paths are
+     * {@linkplain Path#resolve(Path) resolved} against the given path.
+     *
+     * @param dir the {@code Path} to list
      *
-     * @throws IOException If an I/O error occurs while accessing the path
+     * @throws IOException if an I/O error occurs while listing entries
      */
-    public static List<Path> listDirectory(Path directoryPath) throws IOException {
-        try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(directoryPath)) {
-            return Lists.newArrayList(directoryStream);
+    public static List<Path> listDirectoryFiles(Path dir) throws IOException {
+        try (DirectoryStream<Path> entries = Files.newDirectoryStream(dir, FILE_FILTER)) {
```

Comment:
how do we feel about using the public method `regularFileFIlter` in place of referencing the constant directly?

It might make it easier for users to copy our approach in their own libraries.